From 696c616049e56a0667cfc7651ac89db343e0f4ee Mon Sep 17 00:00:00 2001
From: Eli Barzilay <eli@barzilay.org>
Date: Fri, 3 Dec 2021 12:12:23 -0500
Subject: [PATCH 5/5] Add a bundling step for tests

This will change (should be done for all packages, and probably better),
but added now to make running the tests easier for others.
---
 Gulpfile.js | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/Gulpfile.js b/Gulpfile.js
index edf1cebb00..79f4273715 100644
--- a/Gulpfile.js
+++ b/Gulpfile.js
@@ -18,8 +18,9 @@ const { buildProject, cleanProject, watchProject } = require("./scripts/build/pr
 const cmdLineOptions = require("./scripts/build/options");
 
 const copyright = "CopyrightNotice.txt";
-const testRoot = "built/local/testRunner/Harness.js";
-//!!! const testRoot = "built/local/run.js";
+const testRootFile = "built/local/testRunner/Harness.js";
+const testRootBundle = "built/local/run.js";
+const testRoot = testRootBundle; // use testRootFile to run directly from the module files
 const cleanTasks = [];
 
 const buildScripts = () => buildProject("scripts");
@@ -322,7 +323,14 @@ task("watch-lssl").flags = {
     "   --built": "Compile using the built version of the compiler."
 };
 
-const buildTests = () => buildProject("src/testRunner");
+const buildTests = series(
+  () => buildProject("src/testRunner"),
+  () => exec("npx", [
+    "esbuild", "--bundle", "--format=cjs", "--platform=node", "--target=node12",
+    "--legal-comments=none",
+    `--outfile=${testRootBundle}`, testRootFile,
+  ]),
+);
 task("tests", series(preBuild, parallel(buildLssl, buildTests)));
 task("tests").description = "Builds the test infrastructure";
 task("tests").flags = {
-- 
2.33.1

